src/providers/pubsub.ts

```diff
@@ -107,7 +102,7 @@ export function _scheduleWithOptions(
   schedule: string,
   options: DeploymentOptions
 ): ScheduleBuilder {
-  return new ScheduleBuilder({ schedule }, options);
+  return new ScheduleBuilder({ ...options, schedule: { schedule } });
```

Comment:
Apparently `schedule` of type `object` has a property `schedule` of type `string`. Whereas it might look wrong at a first glance, that's how it's defined.

Reply:
Good catch, yeah it's very nested and we should probably just rename this instead of having schedule => Schedule => schedule (but don't worry about that for now). Also missed in the code review that DeploymentOptions already had schedule in it. You have very good attention to detail!